AmplifyPlugins/Analytics/AWSPinpointAnalyticsPluginTests/AnalyticsEventStorageTests.swift

```diff
@@ -70,38 +82,261 @@ class AnalyticsEventStorageTests: XCTestCase {
+            
+            try storage.deleteDirtyEvents()
+            
+            eventcount = try adapter.executeQuery(eventCountStatement, []).scalar() as! Int64
+            dirtyEventcount = try adapter.executeQuery(dirtyEventCountStatement, []).scalar() as! Int64
             XCTAssertTrue(eventcount == 2)
+            XCTAssertTrue(dirtyEventcount == 0)
+        } catch {
+            XCTFail("Failed to delete all dirty events")
+        }
+    }
+    
+    /// - Given: a local storage
+    /// - When: delete oldest event is called
+    /// - Then: the oldest event in the event table is removed
+    func testDeleteOldestEvent() {
+        do {
+            var eventcount = try adapter.executeQuery(eventCountStatement, []).scalar() as! Int64
+            var dirtyEventcount = try adapter.executeQuery(dirtyEventCountStatement, []).scalar() as! Int64
+            XCTAssertTrue(eventcount == 3)
             XCTAssertTrue(dirtyEventcount == 2)
+
+            try storage.deleteOldestEvent()
+
+            eventcount = try adapter.executeQuery(eventCountStatement, []).scalar() as! Int64
+            dirtyEventcount = try adapter.executeQuery(dirtyEventCountStatement, []).scalar() as! Int64
+            XCTAssertTrue(eventcount == 2)
+            XCTAssertTrue(dirtyEventcount == 2)
+
+            let events = try storage.getEventsWith(limit: 5)
+            XCTAssertTrue(events.count == 2)
+            XCTAssertTrue(events.contains(where: { $0.id ==  "1"}))
+            XCTAssertTrue(events.contains(where: { $0.id ==  "2"}))
+        } catch {
+            XCTFail("Failed to delete oldest event")
+        }
+    }
+    
+    /// - Given: a local storage
+    /// - When: get a list of of events given a limit is called
+    /// - Then: a list of events is returned
+    func testGetEventsWithLimit() {
+        do {
+
+            let events = try storage.getEventsWith(limit: 5)
+            XCTAssertTrue(events.count == 3)
+            XCTAssertTrue(events.contains(where: { $0.id ==  "1"}))
+            XCTAssertTrue(events.contains(where: { $0.id ==  "2"}))
+            XCTAssertTrue(events.contains(where: { $0.id ==  "5"}))
+        } catch {
+            XCTFail("Failed to get events")
+        }
+    }
+    
+    /// - Given: a local storage
+    /// - When: get events with limit is called
+    /// - Then: a event with the list of event has expect properties and attributes
+    func testPinpointEventConversion() {
+        do {
+            let expectedSessionStartTime = DateFormatter.iso8601DateFormatterWithFractionalSeconds.date(from: "2022-06-10T17:00:20.618+0000")
+            let expectedSessionStopTime = DateFormatter.iso8601DateFormatterWithFractionalSeconds.date(from: "2022-06-10T17:10:20.618+0000")
+            let events = try storage.getEventsWith(limit: 5)
+            let latestEvent = events.first(where: { $0.id == "1" })
+            XCTAssertNotNil(latestEvent)
+            XCTAssertTrue(latestEvent?.eventType == "eventType")
```

Comment:
nit: This is perhaps personal preference, but I recommend using `XCTAssertEqual` on all equality validations, as the error messages will be more descriptive on failure.